haystack/components/preprocessors/utils.py

```diff
@@ -0,0 +1,231 @@
+            language=language,
+        )
+        sentence_tokenizer = nltk.data.load("tokenizers/punkt_tab/english.pickle")
+
+    if keep_white_spaces:
+        sentence_tokenizer._lang_vars = CustomPunktLanguageVars()
+
+    return sentence_tokenizer
+
+
+class SentenceSplitter:  # pylint: disable=too-few-public-methods
+    """
+    SentenceSplitter splits a text into sentences using the nltk sentence tokenizer
+    """
+
+    def __init__(
+        self,
+        language: Language = "en",
+        use_split_rules: bool = True,
+        extend_abbreviations: bool = True,
+        keep_white_spaces: bool = False,
+    ) -> None:
+        """
+        Initializes the SentenceSplitter with the specified language, split rules, and abbreviation handling.
+
+        :param language: The language for the tokenizer. Default is "en".
+        :param use_split_rules: If True, the additional split rules are used. If False, the rules are not used.
+        :param extend_abbreviations: If True, the abbreviations used by NLTK's PunktTokenizer are extended by a list
+            of curated abbreviations if available. If False, the default abbreviations are used.
+            Currently supported languages are: en, de.
+        :param keep_white_spaces: If True, the tokenizer will keep white spaces between sentences.
+        """
+        self.language = language
+        self.sentence_tokenizer = load_sentence_tokenizer(language, keep_white_spaces=keep_white_spaces)
+        self.use_split_rules = use_split_rules
+        if extend_abbreviations:
+            abbreviations = SentenceSplitter._read_abbreviations(language)
+            self.sentence_tokenizer._params.abbrev_types.update(abbreviations)
+        self.keep_white_spaces = keep_white_spaces
+
+    def split_sentences(self, text: str) -> List[Dict[str, Any]]:
+        """
+        Splits a text into sentences including references to original char positions for each split.
+
+        :param text: The text to split.
+        :returns: list of sentences with positions.
+        """
+        sentence_spans = list(self.sentence_tokenizer.span_tokenize(text))
+        if self.use_split_rules:
+            sentence_spans = SentenceSplitter._apply_split_rules(text, sentence_spans)
+
+        sentences = [{"sentence": text[start:end], "start": start, "end": end} for start, end in sentence_spans]
+        return sentences
+
+    @staticmethod
+    def _apply_split_rules(text: str, sentence_spans: List[Tuple[int, int]]) -> List[Tuple[int, int]]:
+        """
+        Applies additional split rules to the sentence spans.
+
+        :param text: The text to split.
+        :param sentence_spans: The list of sentence spans to split.
+        :returns: The list of sentence spans after applying the split rules.
+        """
+        new_sentence_spans = []
+        quote_spans = [match.span() for match in re.finditer(r"\W(\"+|\'+).*?\1", text)]
```

Comment:
If this is going to be used hundreds or thousands of times for a collection of documents, the regular expression could be compiled only once and then used multiple times, which also helps separate the definition of the regex form it's use.

Either at the top of the file defining something like `QUOTE_SPANS_RE=re.compile("\W(\"+|\'+).*?\1")` or in the `init()` of this class and then here `re.finditer(QUOTE_SPANS_RE)`